worker/sshserver/server.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/juju/juju/state"
+)
+
+// NewSSHServer returns an embedded SSH server. This server does a few things,
+// reuqests that come in must go through the jump server. The jump server will
+// pipe the connection and pass it into an in-memory instance of another SSH server.
+//
+// This second SSH server (seen within directTCPIPHandlerClosure) will handle the
+// the termination of the SSH connections, note, it is not listening on any ports
+// because we are passing the piped connection to it, essentially allowing the following
+// to work (despite only having one server listening):
+// - `ssh -J controller:2223 ubuntu@app.controller.model`
```

Comment:
These comments could go into the worker doc.go as a general explanation.

Reply:
Moving it to doc.go